Public/Src/FrontEnd/Nuget/NugetSpecGenerator.cs

```diff
@@ -115,68 +119,80 @@ public ISourceFile CreatePackageConfig()
                     {
-                        runtime.Add(CreateSimpleBinary(assembly));
+                        foreach (var assembly in libAssemblies)
+                        {
+                            runtime.Add(CreateSimpleBinary(assembly));
+                        }
                     }
-                }
 
-                // Dependency items
-                if (analyzedPackage.DependenciesPerFramework.TryGetValue(
-                    framework.Key,
-                    out IReadOnlyList<INugetPackage> dependencySpecificFrameworks))
-                {
-                    foreach (var dependencySpecificFramework in dependencySpecificFrameworks)
+                    // Dependency items
+                    if (analyzedPackage.DependenciesPerFramework.TryGetValue(monikers.First(), out IReadOnlyList<INugetPackage> dependencySpecificFrameworks))
                     {
-                        dependencies.Add(CreateImportFromForDependency(dependencySpecificFramework));
+                        foreach (var dependencySpecificFramework in dependencySpecificFrameworks)
+                        {
+                            dependencies.Add(CreateImportFromForDependency(dependencySpecificFramework));
+                        }
                     }
-                }
 
-                dependencies.AddRange(analyzedPackage.Dependencies.Select(CreateImportFromForDependency));
-
-                cases.Add(
-                    new CaseClause(
-                        new LiteralExpression(framework.Key.ToString(m_pathTable.StringTable)),
-                        new ReturnStatement(
-                            new CallExpression(
-                                PropertyAccess("Managed", "Factory", "createNugetPackge"),
-                                new LiteralExpression(analyzedPackage.Id),
-                                new LiteralExpression(analyzedPackage.Version),
-                                PropertyAccess("Contents", "all"),
-                                Array(compile),
-                                Array(runtime),
-                                Array(dependencies)
+                    cases.Add(
+                        new CaseClause(
+                            new LiteralExpression(monikers.Last().ToString(m_pathTable.StringTable)),
+                            new ReturnStatement(
+                                new CallExpression(
+                                    PropertyAccess("Managed", "Factory", "createNugetPackage"),
+                                    new LiteralExpression(analyzedPackage.Id),
+                                    new LiteralExpression(analyzedPackage.Version),
+                                    PropertyAccess("Contents", "all"),
+                                    Array(compile),
+                                    Array(runtime),
+                                    Array(new CallExpression(new Identifier("...addIfLazy"),

```

Comment:
>("... [](start = 75, length = 5)

I fully don't understand why we generate the dependnecies only for the first one.
Is this to emulate that we are still generating the code wrong.
i.e. the compatiblity fallback logic should technically be done on each individual aspect:
* runtime
* compile
* dependencies

Reply:
We don't generate only for the first one, but for every assembly in the package up to all the versions higher (compatible) or the next assembly with a higher tfm version. Here is an example of the a generated spec:

```
@@public
export const pkg: Managed.ManagedNugetPackage = (() => {
    switch (qualifier.targetFramework) {
        case "net20":
        case "net35":
            return Managed.Factory.createNugetPackage(
                "System.Data.SQLite.Linq",
                "1.0.102.0",
                Contents.all,
                [Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net20/System.Data.SQLite.Linq.dll`)],
                [
                    Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net20/System.Data.SQLite.Linq.dll`),
                ],
                [...addIfLazy(qualifier.targetFramework === "net20", () => [])]
            );
        case "net40":
            return Managed.Factory.createNugetPackage(
                "System.Data.SQLite.Linq",
                "1.0.102.0",
                Contents.all,
                [Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net40/System.Data.SQLite.Linq.dll`)],
                [
                    Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net40/System.Data.SQLite.Linq.dll`),
                ],
                [...addIfLazy(qualifier.targetFramework === "net40", () => [])]
            );
        case "net45":
            return Managed.Factory.createNugetPackage(
                "System.Data.SQLite.Linq",
                "1.0.102.0",
                Contents.all,
                [Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net45/System.Data.SQLite.Linq.dll`)],
                [
                    Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net45/System.Data.SQLite.Linq.dll`),
                ],
                [...addIfLazy(qualifier.targetFramework === "net45", () => [])]
            );
        case "net451":
        case "net452":
            return Managed.Factory.createNugetPackage(
                "System.Data.SQLite.Linq",
                "1.0.102.0",
                Contents.all,
                [Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net451/System.Data.SQLite.Linq.dll`)],
                [
                    Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net451/System.Data.SQLite.Linq.dll`),
                ],
                [...addIfLazy(qualifier.targetFramework === "net451", () => [])]
            );
        case "net46":
        case "net461":
        case "net462":
        case "net472":
            return Managed.Factory.createNugetPackage(
                "System.Data.SQLite.Linq",
                "1.0.102.0",
                Contents.all,
                [Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net46/System.Data.SQLite.Linq.dll`)],
                [
                    Managed.Factory.createBinaryFromFiles(f`${packageRoot}/lib/net46/System.Data.SQLite.Linq.dll`),
                ],
                [...addIfLazy(qualifier.targetFramework === "net46", () => [])]
            );
        default:
            Contract.fail("Unsupported target framework");
    };
}
)();
```